app/models/concerns/environmental.rb

```diff
@@ -153,6 +156,10 @@ def has_environment(default: nil)
             next unless
               association.class < Environmental
 
+            # See: https://github.com/rails/rails/issues/51853
+            environment_id = read_attribute(:environment_id) ||
+                             environment&.id
```

Comment:
This is a hack for https://github.com/rails/rails/issues/51853

Reply:
Doesn't work either. I think I'm blocked by the change in order of operations right now. 😩

Gonna see if I can figure out how to fix upstream instead of trying workarounds.